packages/expo-manifests/ios/EXManifests/EXManifestsBaseManifest.m

```diff
@@ -20,7 +20,7 @@ - (NSString *)description {
 # pragma mark - Field Getters
 
 - (NSString *)legacyId {
-  return [self.rawManifestJSON nullableStringForKey:@"id"];
+  return [self.rawManifestJSON stringForKey:@"id"];
```

Comment:
Let's make the same change in all the past versions of this as well (`ABI40_0_0EXManifestsBaseManifest`, ...).

For this library we've been keeping the versions in sync to mimic the behavior in android (they're unversioned in android) so that there are less discrepancies when trying to figure out what's going wrong.